File: javascript/atoi.js
function atoi(str) {
  let i = 0;
  let sign = 1;
  let result = 0;

  // Skip leading whitespace
  while (i < str.length && str[i] === ' ') i++;

  // Check for sign
  if (str[i] === '+' || str[i] === '-') {
    sign = (str[i] === '-') ? -1 : 1;
    i++;
  }

  // Convert digits
  while (i < str.length && str[i] >= '0' && str[i] <= '9') {
    result = result * 10 + (str.charCodeAt(i) - '0'.charCodeAt(0));
    i++;
  }

  return sign * result;
}

// Example usage:
console.log(atoi("   -12345")); // Output: -12345
console.log(atoi("42"));        // Output: 42

File: Python/atoi.py
def atoi(s: str) -> int:
    i = 0
    sign = 1
    result = 0
    n = len(s)
    
    # Skip leading spaces
    while i < n and s[i].isspace():
        i += 1
    
    # Check sign
    if i < n and (s[i] == '+' or s[i] == '-'):
        sign = -1 if s[i] == '-' else 1
        i += 1
    
    # Convert digits
    while i < n and s[i].isdigit():
        result = result * 10 + int(s[i])
        i += 1
    
    return sign * result

# Example usage:
print(atoi("   -789"))  # Output: -789
print(atoi("+42"))      # Output: 42
